### PR TITLE
[SPARK-45603][INFRA] merge_spark_pr shall notify us about GITHUB_OAUTH_KEY expiry

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR handles unauthorized errors when calling github rest API to notify us about the 
GITHUB_OAUTH_KEY expiry as the current hint does not help

```
git rev-parse --abbrev-ref HEAD
Unable to fetch URL, exiting: https://api.github.com/repos/apache/spark/branches
Restoring head pointer to master
```

### Why are the changes needed?

improve infra scripts

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

```shell
./dev/merge_spark_pr.py
git rev-parse --abbrev-ref HEAD
GITHUB_OAUTH_KEY is invalid or expired. Please regenerate a new one with at least the 'public_repo' scope on https://github.com/settings/tokens and update your local settings before you try again.
Restoring head pointer to master
```

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: dev/merge_spark_pr.py [hunk 1]
```diff
@@ -85,6 +85,12 @@ def get_json(url):
                 + "dev/merge_spark_pr.py to configure an OAuth token for making authenticated "
                 + "GitHub requests."
             )
+        elif e.code == 401:
```
Comment (HyukjinKwon) [inline dev/merge_spark_pr.py:88]: Hm, is 401 always about GITHUB_OAUTH_KEY expiry?

Comment (yaooqinn) [reply dev/merge_spark_pr.py:88]: In Github rest APIs, 401 stands for `requires authentication`. The token could be invalid with expiry or wrong configuration

**Review by HyukjinKwon — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Review by beliefer — COMMENTED** [review]
LGTM later.

**Pull request closed without merging**